### PR TITLE
Logging context unit tests

### PR DESCRIPTION
**Pull request by HardNorth**

### CODE REVIEW

File: src/test/java/com/epam/reportportal/test/TestUtils.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package com.epam.reportportal.test;
```
Comment (pbortnik) [inline src/test/java/com/epam/reportportal/test/TestUtils.java]: Header.
Maybe you can configure auto header for new files.

File: src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		final List<Future<String>> results = pooledExecutor.invokeAll(Arrays.asList(t1, t2));
+
+		Awaitility.await("Wait until test finish").until(() -> results.stream().filter(Future::isDone).collect(Collectors.toList()), hasSize(2));
+
+		// Verify all item start requests passed
+		verify(rpClient, times(1)).startLaunch(any());
+		verify(rpClient, times(1)).startTestItem(any());
+		verify(rpClient, times(2)).startTestItem(anyString(), any());
+
+		// Verify 2 log are logged and save their requests
```
Comment (Yumfriez) [inline src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java]: 10 logs, probably

File: src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+		simulateStartLaunchResponse(rpClient);
+		simulateStartTestItemResponse(rpClient);
+		simulateStartChildTestItemResponse(rpClient);
+		simulateBatchLogResponse(rpClient);
+		rp = new ReportPortal(rpClient, clientExecutorService, params, null);
+	}
+
+	@After
+	public void cleanUp() {
+
```
Comment (ihar-kahadouski) [inline src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java]: Do we need empty method?

File: src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java (hunk 2 above)
Comment (HardNorth) [reply src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java]: Updated

File: src/test/java/com/epam/reportportal/test/TestUtils.java (hunk 1 above)
Comment (HardNorth) [reply src/test/java/com/epam/reportportal/test/TestUtils.java]: Added.

File: src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java (hunk 3 above)
Comment (HardNorth) [reply src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java]: Removed

**Pull request merged**